### PR TITLE
haskell: optional hindent-mode

### PR DESCRIPTION
**Pull request by michelk**

### CODE REVIEW

**Comment by bjarkevad** [comment]
Does this conflict with `structured-haskell-mode`? Also `m f` is already bound to haskell stylish, so `m f q` will not work.

**Comment by michelk** [comment]
As mentioned [here](https://github.com/chrisdone/emacs-haskell-config), `hindent` and `shm` shouldn't conflict. This config might also serve as a blueprint for the spacemacs-haskell-configuration.

I just changed the keybinding to `m g q`, which is similar to VIM-reformating, but doesn't really fit into `spacemacs` conventions. Any good idea?

**Comment by bjarkevad** [comment]
Ah cool! I'll have a look at the config when I have some time. 

Maybe put it on `m F`?

**Comment by michelk** [comment]
`m F` sounds like a good idea; changed it.

File: contrib/lang/haskell/packages.el [hunk 1]
```diff
@@ -18,9 +18,14 @@
     ghc
     haskell-mode
     hi2
-    shm
```
Comment (syl20bnr) [inline contrib/lang/haskell/packages.el]: Is it intended ?

Comment (bjarkevad) [inline contrib/lang/haskell/packages.el]: I doubt it

File: contrib/lang/haskell/packages.el [hunk 2]
```diff
@@ -18,9 +18,14 @@
...
     ))
 
+(when haskell-enable-shm-support
+  (add-to-list 'haskell-packages 'shm))
```
Comment (michelk) [inline contrib/lang/haskell/packages.el]: I'm adding it optionally.

**Comment by syl20bnr** [comment]
Oops sorry, I did not see it.
I moved the packages declaration in `haskell-packages` though because it makes them discoverable with `SPC f e h`, the side effect is that they are always installed with the haskell layer but it is not a problem.

Thank you !
Cherry-picked into `develop`, you can safely delete your branch.

**Comment by syl20bnr** [comment]
See commit a024d7452112e42d89ff5f9315070a33d941af19 for the modification mentioned above.

**Pull request closed without merging**